client/session.go

```diff
@@ -1158,6 +1211,8 @@ func (s *session) fetchTaggedAttemptWithRLock(
 		fetchState = s.pools.fetchState.Get()
 	)
 
+	fetchState.nsID = nsID // transfer ownership of nsID to `fetchState`
```

Comment:
Any reason not to clone here instead?  Might be cleaner, that way you don't have to clone in two places.

Reply:
Yep, the value returned by `convert.ToRPCFetchTaggedRequest(nsClone, ...)` takes reference to the bytes underlying the `nsID` clone, so the nsID clone has to precede the request creation. 

I'm doing the RPC request creation early (before fetchState/Op creation) to avoid the extra allocs in case of failures.